### PR TITLE
Reduce loading of AILTN events at app background

### PR DESCRIPTION
**Pull request by bcriscuolo**
Based on work necessary for Salesforce Field Service, this PR encompasses work to enable batch processing of AILTN events, as well as strategic usage of `@autoreleasepool` blocks to more effectively manage runtime memory allocations while processing.

Doing so significantly reduces the memory profile of the host application when processing AILTN events stored at runtime.

As you can see below, the biggest improvement comes from autorelease support, with batching being a more effective means to attempt to submit these metrics before a background task terminates (fewer submissions, although concurrently, may succeed better than one giant one).

Memory analysis

Test 1:
1000 events (single "batch")
Autorelease disabled
AILTN start: 445Mb
AILTN end: 1.25Gb <------

Test 2:
1000 events (single "batch")
Autorelease enabled
AILTN start: 443Mb
AILTN end: 451Mb

Test 3:
1000 events, processed 100 x 10
Autorelease disabled
AILTN start: 437Mb
AILTN end: 447Mb

Test 4:
1000 events, processed 100 x 10
Autorelease enabled
AILTN start: 433Mb
AILTN end: 441Mb

### CODE REVIEW

**Comment by bcriscuolo** [comment]
@bbirman

**Review by bbirman — APPROVED** [review]
thank you!

**Review by brandonpage — APPROVED** [review]

**Review by wmathurin — APPROVED** [review]

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.h [hunk 1]
```diff
@@ -116,4 +124,10 @@
  */
 - (void) updateLoggingPrefs;
 
+
+/**
+ * Clears unpublished events.
+ */
+- (void) clearAllEvents;
```
Comment (bbirman) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.h:131]: @bcriscuolo sorry one more question, I can't find the method definition for this one?

Comment (bcriscuolo) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.h:131]: That is unnecessary, my apologies. I will remove this ASAP from this interface (SFS-iOS uses the `deleteAllEvents` method).

**Pull request merged**